src/rhub/bare_metal/tasks/host.py

```diff
@@ -0,0 +1,77 @@
+import logging
+
+from rhub.api import db
+from rhub.api.extensions import celery
+from rhub.bare_metal.model import (
+    BareMetalException,
+    BareMetalHost,
+    BareMetalHostDrac,
+    BareMetalHostRedfish,
+    BareMetalHostStatus,
+    BareMetalIronicHandler,
+    IronicNode,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def return_host_to_pool(host: BareMetalHost):
+    host.status = BareMetalHostStatus.AVAILABLE
+    db.session.commit()
+
+
+@celery.task()
+def ironic_enroll_host_task(host_id):
+    host: BareMetalHost = BareMetalHost.query.get(host_id)
+
+    if not host.can_be_enrolled():
+        raise RuntimeError(f"Host {host} can't be enrolled")
+
+    if not isinstance(host.handler, BareMetalIronicHandler):
+        raise NotImplementedError()
+
+    ironic: BareMetalIronicHandler = host.handler
+
+    logger.info(f"Enrolling {host} on {ironic}")
+    host.status = BareMetalHostStatus.ENROLLING
+    db.session.commit()
+
+    node_creation_data = dict(
+        # TODO: discuss automated_clean - no need to disable on yoga
+        automated_clean=False,
+        boot_interface="ipxe",
```

Comment:
I just worked on a machine that only supported PXE so this may need to be configurable.